### PR TITLE
B 20757 int

### PR DESCRIPTION
**Pull request by CoryKleinjanCACI**
## Summary

This task updates the error message for DTOD so it is clear why the PPM cannot progress. Follow-on tasks will provide a more granular error message based on reason for failure.

## Verification Steps for the Author

These are to be checked by the author.

- [ ] Tested in the Experimental environment (for changes to containers, app startup, or connection to data stores)
- [ ] Have the Agility acceptance criteria been met for this change?

## Verification Steps for Reviewers

These are to be checked by a reviewer.

- [ ] Has the branch been pulled in and checked out?
- [ ] Have the BL acceptance criteria been met for this change?
- [ ] Was the CircleCI build successful?
- [ ] Has the code been reviewed from a standards and best practices point of view?

### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/getting-started/application-setup/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/getting-started/development/testing)

### How to test

NOTE: Prior to test I recommend adding "distanceFloat = -1" after line 121 of dtod_zip5_distance.go to simulate a DTOD failure.

1. Access the milmove application
2. Login as a Customer
3. Create a PPM
4. Fill in all field for the ppm and progress to the estimated incentives page (use bad zips if you have them, refer to testing note for guidance on how to simulate failures)
5. Fill in estimated weight
6. Save and Continue
7. Verify that error message matches the error message in the acceptance criteria for [B-20757](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3A983881)

### Frontend

- [ ] There are no aXe warnings for UI.
- [ ] This works in [Supported Browsers and their phone views](https://transcom.github.io/mymove-docs/docs/adrs/Browser-Support/#minimum-browser-requirements) (Chrome, Firefox, Edge).
- [ ] There are no new console errors in the browser devtools.
- [ ] There are no new console errors in the test output.
- [ ] If this PR adds a new component to Storybook, it ensures the component is fully responsive, OR if it is intentionally not, a wrapping div using the `officeApp` class or custom `min-width` styling is used to hide any states the would not be visible to the user.
- [ ] This change meets the standards for [Section 508 compliance](https://www.ssa.gov/accessibility/andi/help/install.html).

### Backend

- [ ] Code follows the guidelines for [Logging](https://transcom.github.io/mymove-docs/docs/getting-started/development/logging).
- [ ] The requirements listed in [Querying the Database Safely](https://transcom.github.io/mymove-docs/docs/backend/guides/golang-guide#querying-the-database-safely) have been satisfied.

#### Any new migrations/schema changes:

- [ ] Follows our guidelines for [Zero-Downtime Deploys](https://transcom.github.io/mymove-docs/docs/backend/setup/database-migrations#zero-downtime-migrations).
- [ ] Have been communicated to #g-database.
- [ ] Secure migrations have been tested following the instructions in our [docs](https://transcom.github.io/mymove-docs/docs/backend/setup/database-migrations#secure-migrations).

## Screenshots

### CODE REVIEW

**Review by ajlusk — CHANGES_REQUESTED** [review]
Acceptance Criteria states: 'The words "Technical Help Desk"  should be a hyperlink to contact the help desk via email: {email address that I don't know if I should post on github}'

Screenshot 2024-10-01 at 11 57 20 AM

Also, it might be necessary to add a test for this scenario since we've been teetering on the acceptable line for test coverage. Gets this closer to its original 5pts at least.

**Comment by r-mettler** [comment]
> Acceptance Criteria states: 'The words "Technical Help Desk"  should be a hyperlink to contact the help desk via email: {email address that I don't know if I should post on github}'
> 
> Screenshot 2024-10-01 at 11 57 20 AM
> 
> Also, it might be necessary to add a test for this scenario since we've been teetering on the acceptable line for test coverage. Gets this closer to its original 5pts at least.

I would add the a constant like I had in B-19800 but that was pulled back. I think we want to use the URL for the help desk because the email address may change. See

File: src/pages/MyMove/PPM/Booking/EstimatedWeightsProGear/EstimatedWeightsProGear.jsx [hunk 1]
```diff
@@ -74,7 +77,14 @@ const EstimatedWeightsProGear = () => {
             <h1>Estimated weight</h1>
             {errorMessage && (
               <Alert headingLevel="h4" slim type="error">
-                {errorMessage}
+                {errorCode === 400 || errorCode === 500 ? (
+                  <p>
```
Comment (r-mettler) [inline src/pages/MyMove/PPM/Booking/EstimatedWeightsProGear/EstimatedWeightsProGear.jsx:81]: Can you change the link to the helpdesk so it pops open in a new tab.
`<a href={technicalHelpDeskURL} target="_blank" rel="noreferrer">
     Technical Help Desk
 </a>`

**Comment by ajlusk** [comment]
> # Test coverage on server failed.
> **Last run: Wed 02 Oct 2024 02:57:02 AM UTC** Refer to [this report](https://output.circle-artifacts.com/output/job/3476ffc8-72f5-49bd-9e8f-c8b14c55a2a8/artifacts/0/test-results/gotest/app/go-coverage.html) to see coverage details for the server app. Documentation for test coverage can be found [at this link](https://transcom.github.io/mymove-docs/docs/about/development/test-coverage-reports).

My suggestion to add testing was more about the backend handler. Testing that it return the expected error for bad zips or a simulated outage.

**Review by r-mettler — APPROVED** [review]
Looks good, opens link in new tab now.

**Comment by r-mettler** [comment]
> > # Test coverage on server failed.
> > **Last run: Wed 02 Oct 2024 02:57:02 AM UTC** Refer to [this report](https://output.circle-artifacts.com/output/job/3476ffc8-72f5-49bd-9e8f-c8b14c55a2a8/artifacts/0/test-results/gotest/app/go-coverage.html) to see coverage details for the server app. Documentation for test coverage can be found [at this link](https://transcom.github.io/mymove-docs/docs/about/development/test-coverage-reports).
> 
> My suggestion to add testing was more about the backend handler. Testing that it return the expected error for bad zips or a simulated outage.

I believe this was showing because of the way some of our tests were setup they always fail on the first of the month. Cameron resolved this with a [PR fix](https://github.com/transcom/mymove/pull/13807) he put in yesterday. But it wouldn't hurt to add some tests for bad zip for this as suggested.

**Review by r-mettler — APPROVED** [review]

**Review by ajlusk — APPROVED** [review]
Updates made from review suggestions look good!

**Pull request merged**